stdlib/2and3/distutils/version.pyi

```diff
@@ -1,31 +1,38 @@
 import sys
-from typing import Any
+from typing import Any, Optional, TypeVar, Union, Pattern, Tuple
+
+T = TypeVar('T', bound='Version')
 
 class Version:
-    def __init__(self, vstring=None): ...
+    def __init__(self, vstring: Optional[str] = None) -> None: ...
+    def __repr__(self) -> str: ...
 
     if sys.version_info >= (3,):
-        def __eq__(self, other): ...
-        def __lt__(self, other): ...
-        def __le__(self, other): ...
-        def __gt__(self, other): ...
-        def __ge__(self, other): ...
+        def __eq__(self, other: object) -> bool: ...
+        def __lt__(self: T, other: Union[T, str]) -> bool: ...
+        def __le__(self: T, other: Union[T, str]) -> bool: ...
+        def __gt__(self: T, other: Union[T, str]) -> bool: ...
+        def __ge__(self: T, other: Union[T, str]) -> bool: ...
+
+    # Methods part of the 'Version' interface, but not implemented
+    # within the class itself
```

Comment:
Isn't the recommended pattern here to use @abstractmethod in the base class and add concrete methods to the subclasses? (Or, as a compromise, make only `__init__` abstract and add just a concrete `__init__` to the subclasses.) Since presumably you can't instantiate `Version`.

Reply:
That's true -- the main reason I didn't do that is because the actual implementations of version.py don't import and use the abc module or use the `@abstractmethod` decorator.

Should I go ahead and do that within the stubs? I don't quite remember if there are any issues with having just the stubs use AbcMeta as a metaclass when the concrete implementations don't.